src/doc/unstable-book/src/language-features/string-deref-patterns.md

```diff
@@ -0,0 +1,45 @@
+# `string_deref_patterns`
+
+The tracking issue for this feature is: [#87121]
+
+[#87121]: https://github.com/rust-lang/rust/issues/87121
+
+------------------------
+
+This feature permits pattern matching `String` to `&str` through [its `Deref` implementation].
+
+```rust
+#![feature(string_deref_patterns)]
+
+pub enum Value {
+    String(String),
+    Number(u32),
+}
+
+pub fn is_it_the_answer(value: Value) -> bool {
+    match value {
+        Value::String("42") => true,
+        Value::Number(42) => true,
+        _ => false,
+    }
+}
+```
+
+Without this feature other constructs such as an "if let guard" have to be used.
```

Comment:
An if-let guard would be `Value::String(s) if let "42" = &s => … which is still unstable (`if_let_guard`) and probably shouldn't be mentioned here. What your snippet contains is an ordinary [match guard](https://doc.rust-lang.org/stable/reference/expressions/match-expr.html#match-guards).

```suggestion
Without this feature other constructs such as match guards have to be used.
```

Reply:
Oops, thanks. I originally had another example but forget to update the text 😆